developers/release.md

```diff
@@ -0,0 +1,348 @@
+
+### Step 13: Test
+
+Use the newly updated katello-deploy to spin up an X.Y box and test for the following:
+
+* Installation
+* UI
+* Repository sync and content view publish
+* Can install katello-agent for el5,6, 7 from the client repo (without the need of the main katello repo enabled)
+* Run a full bats test with content
+
+### Step 14: Announce RC
+
+Now you are ready to announce the RC by sending an email to the foreman-users and foreman-dev mailing lists pointing users at the installation instructions. You should also open a PR to `katello.org` to update the news section with an announcement of the RC release.
+
+### Step 15: Fix Bugs
+
+Start off by creating a label that represents the X.Y version, e.g. `KATELLO-2.1` on the relevant repositories. The workflow until release will be as follows: 
+
+1. Mark any open pull request that should go into X.Y with 'Needs Backporting' and 'Katello X.Y'.
+1. Periodically go through and backport all the merged 'needs backporting' PRs to the X.Y branch and open one new PR against the X.Y branch with all the backports
+1. Remove the 'needs backporting' label from the PRs after cherry picking
+1. After generated 'backport' is merged, tag and release, mash, and copy-repos to fedorahosted
+
+### Step 16: Release!
+
+After iterating through Step 15 a few times over the course of roughly a month, and in sync with Foreman's release date decide that you are ready to release.
+
+```
+cd katello-misc/repos
+sed -i "s#/gpgcheck=0/#/gpgcheck=1/#g" katello*.repo
+git commit -a -m 'Turning on GPG key checking for X.Y'
+tito tag
+tito release koji
+```
+
+##### Download Unsigned RPMs
+
+Now sign the RPMs (or ask someone who can to do this part for you), mash and release. Those you can ask are lzap, mmccune, jsherrill, bkearney. 
+
+The first step when signing is to download all of the unsigned RPMs from tags that you will be releasing. We are using the .src.rpm to decide if the build as whole is signed or not. For example, if the key was D5A88496:
+
+```
+cd some empty directory
+VERSION=2.3
+for j in rhel5 rhel6 rhel7 fedora20 fedora21 ; do
+  for i in "katello-$VERSION-$j" "katello-$VERSION-thirdparty-candlepin-$j" "katello-$VERSION-thirdparty-pulp-$j" ; do
+    koji -c ~/.koji/katello-config list-tagged --latest --quiet --inherit --sigs $i ; done \
+    | sed 's!^!:!' \
+    | perl -ane '$F[1] =~ s!\.src$!! or next; $R{$F[1]} = 1; $S{$F[1]} = 1 if $F[0] eq ":D5A88496";
+      END { print map "$_\n", grep { not exists $S{$_} } sort keys %R }' \
+    | while read i ; do koji -c ~/.koji/katello-config download-build --debuginfo $i ; 
+  done;
+done;
+```
+
+##### Sign the RPMs
+
+Get (decrypt) Katello key and gpg --import it. Of course, you need to be a person who knows how to decrypt the key. Decrypt the passphrase as well. The key that we've used starting with Katello 1.0 is D5A88496. Before doing the first sign, import katello key. In our git repo with keys do this:
```

Comment:
This Key id is out of date (and the reference to 1.0).   bc62d13f  is the new key id